app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -1,153 +1,271 @@
+import React, { useEffect, useState } from "react";
 import { connect } from "react-redux";
-import { fetchProjectBoard, toggleColumn, reverseColumns } from "actions/projectBoard";
+import { DragDropContext } from "react-beautiful-dnd";
+import {
+  fetchProjectBoard,
+  toggleColumn,
+  reverseColumns
+} from "actions/projectBoard";
 import { fetchPastStories } from "actions/pastIterations";
 import Column from "../Columns/ColumnItem";
 import History from "../stories/History";
 import { getColumns } from "../../selectors/columns";
-import { CHILLY_BIN, DONE, BACKLOG } from '../../models/beta/column';
-import { canCloseColumn } from '../../models/beta/projectBoard';
-import { createStory, closeHistory } from '../../actions/story';
-import AddStoryButton from '../story/AddStoryButton';
-import { status, historyStatus } from 'libs/beta/constants';
-import PropTypes from 'prop-types';
-import StoryPropTypes from '../shapes/story';
-import ProjectBoardPropTypes from '../shapes/projectBoard';
-import Notifications from '../Notifications';
-import { removeNotification } from '../../actions/notifications';
-import StorySearch from '../search/StorySearch';
-import SearchResults from './../search/SearchResults';
-import ProjectLoading from './ProjectLoading';
-import SideBar from './SideBar';
-import Columns from '../Columns';
+import { createStory, closeHistory, dragDropStory } from "../../actions/story";
+import { CHILLY_BIN, DONE, BACKLOG } from "../../models/beta/column";
+import AddStoryButton from "../story/AddStoryButton";
+import { status, historyStatus } from "libs/beta/constants";
+import PropTypes from "prop-types";
+import { storyPropTypesShape } from "../../models/beta/story";
+import {
+  projectBoardPropTypesShape,
+  getNewPosition,
+  getNewSprints,
+  getNewState,
+  moveTask,
+  getBacklogStories,
+  getArray,
+  canCloseColumn
+} from "../../models/beta/projectBoard";
+import StoryPropTypes from "../shapes/story";
+import ProjectBoardPropTypes from "../shapes/projectBoard";
+import Notifications from "../Notifications";
+import { removeNotification } from "../../actions/notifications";
+import StorySearch from "../search/StorySearch";
+import SearchResults from "./../search/SearchResults";
+import ProjectLoading from "./ProjectLoading";
+import SideBar from "./SideBar";
+import Columns from "../Columns";
 import Stories from "../stories/Stories";
 import Sprints from "../stories/Sprints";
 
 export const ProjectBoard = ({
-  fetchProjectBoard,
   projectId,
-  projectBoard,
-  createStory, 
+  createStory,
   closeHistory,
-  notifications, 
-  removeNotification, 
+  notifications,
+  removeNotification,
   history,
+  projectBoard,
+  fetchProjectBoard,
+  dragDropStory,
   chillyBinStories,
   backlogSprints,
   doneSprints,
   fetchPastStories,
   toggleColumn,
   reverseColumns
 }) => {
+  const [newChillyBinStories, setNewChillyBinStories] = useState();
+  const [newBacklogSprints, setNewBacklogSprints] = useState();
```

Comment:
This two const are arrays? if so I think the best way to initialize it is with empty arrays.

```jsx
  const [newChillyBinStories, setNewChillyBinStories] = useState([]);
  const [newBacklogSprints, setNewBacklogSprints] = useState([]);
```

Reply:
Yeah, thanks.